### PR TITLE
fix: Handle 0 safeTxGas when passed to createTransaction call

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves
Resolves [#1343](https://github.com/safe-global/web-core/issues/1343)

If a transaction is created with 0 safeTxGas in the web-core UI and another owner tries to sign it, the `createTransaction` call from the SDK will return a non 0 safeTxGas which causes issues with the existing txData (such as gasLimit estimation failing).

## How this PR fixes it

Narrow the condition to check for `undefined` so that a `safeTxGas` of 0 is caught by it.

### CODE REVIEW

File: packages/safe-core-sdk/tests/createTransaction.test.ts
Comment (mikhailxyz) [inline packages/safe-core-sdk/tests/createTransaction.test.ts:150]: Why this change? For this test it should be irrelevant if it's 0 or 111 ("should return a transaction with defined safeTxGas if safeVersion<1.3.0")

Comment (mikhailxyz) [reply packages/safe-core-sdk/tests/createTransaction.test.ts:150]: I think I get it but i'd a add a separate test for handling falsy values

Comment (usame-algan) [reply packages/safe-core-sdk/tests/createTransaction.test.ts:150]: Yeah, I was thinking its the edge value that caused the issue so rather have a test for it but I will add another.

**Review by mikhailxyz — APPROVED** [review]

**Comment by mikhailxyz** [comment]
wait for a review from @germartinez cos he's the code owner :)

**Review by germartinez — APPROVED** [review]

**Pull request merged**
